package/MDAnalysis/auxiliary/base.py

```diff
@@ -678,8 +772,23 @@ def calc_representative(self):
             except KeyError:
                 value = cutoff_data[min_diff]
         elif self.represent_ts_as == 'average':
-            value = np.mean(np.array([val for val in cutoff_data.values()]),
-                            axis=0)
+            try:
+                value = np.mean(np.array(
+                                [val for val in cutoff_data.values()]
+                                ), axis=0)
+            except TypeError:
```

Comment:
This was previously implemented on the level of the EDRReader, but in a discussion we then agreed to move it back into the base class, which required this check here.

Reply:
🤔 I can't remember the original conversation, however I also can't see this being sustainable, thoughts @orbeckst - I'd personally like to see an issue raise about implementing this as an expected classmethod.

Reply:
I honestly don't remember the conversation, sorry.

I agree with @IAlibay that it does sound like something that's better bundled with the data itself, i.e., *"a method is a function that knows how to operate on the data* so having a `XXXReader._average_data_over_steps()` method would make for cleaner code.

If the only two use cases are either flat numpy arrays or dicts of arrays then it makes sense to keep them here (or as private methods of `base.AUXReader` and re-use the code. 

But if we want to be flexible and keep the actual reader implementations open then just requiring the method is good. We can always provide the flat array method as the method in the base class (so that there are no changes for XVGReader) and start adding it to the individual readers.

I would accept the PR with the current version but would want to see an issue that outlines the required discussion and possible solutions.